Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -77,42 +67,47 @@ struct CombinatorialKalmanFilterTipState {
   std::size_t nHoles = 0;
 };
 
+/// Delegate type that retrieves a range of source links to for a given surface
+/// to be processed by the CKF
+template <typename source_link_iterator_t>
+using SourceLinkAccessorDelegate =
+    Delegate<std::pair<source_link_iterator_t, source_link_iterator_t>(
+        const Surface&)>;
+
 /// Extension struct which holds the delegates to customize the CKF behavior
-template <typename traj_t>
+template <typename source_link_iterator_t, typename traj_t>
 struct CombinatorialKalmanFilterExtensions {
   using candidate_container_t =
       typename std::vector<typename traj_t::TrackStateProxy>;
+
+  using SourceLinkAccessor = SourceLinkAccessorDelegate<source_link_iterator_t>;
+  using Calibrator = typename KalmanFitterExtensions<traj_t>::Calibrator;
+  using Updater = typename KalmanFitterExtensions<traj_t>::Updater;
   using MeasurementSelector =
       Delegate<Result<std::pair<typename candidate_container_t::iterator,
                                 typename candidate_container_t::iterator>>(
           candidate_container_t& trackStates, bool&, const Logger&)>;
   using BranchStopper = Delegate<bool(const CombinatorialKalmanFilterTipState&,
                                       typename traj_t::TrackStateProxy&)>;
 
-  /// The Calibrator is a dedicated calibration algorithm that allows
-  /// to calibrate measurements using track information, this could be
-  /// e.g. sagging for wires, module deformations, etc.
-  typename KalmanFitterExtensions<traj_t>::Calibrator calibrator;
+  /// The source link accessor
+  SourceLinkAccessor sourcelinkAccessor;
```

Comment:
Sorry, for my understanding, why was the `sourcelinkAccessor` moved out of the main `CombinatorialKalmanFilterOptions` into the `CombinatorialKalmanFilterExtensions`?

I suppose it could live in either, but this seems more core-CKF than an extension.